packages/flame_studio/lib/src/run_flame_studio.dart

```diff
@@ -0,0 +1,16 @@
+import 'package:flame_studio/src/widgets/flame_studio.dart';
+import 'package:flutter/widgets.dart';
+
+void runFlameStudio(Widget app, {bool enabled = true}) {
+  var allowRun = true;
+  assert(() {
+    if (enabled) {
```

Comment:
I am trying to understand why this is needed, but I am failing to grasp why, could you go through this so I can get a better idea?

Reply:
So, the idea here is that the user can simply replace `runApp()` with `runFlameStudio()` in their code.

Then, in release mode the `assert()` disappears, and what is left is simple `runApp()`. Thus, in release mode the game will behave exactly as before. Moreover, since this is the only hook into the `flame_studio` package, then the `FlameStudio` and all the internal widgets get eliminated via tree-shaking, so that the size of the final executable is likewise unaffected. 

However, in debug mode the code inside the assert is executed, and that code wraps the user's `app` into the `FlameStudio` widget, and builds the studio UI which controls the game.

Reply:
Ahhh got ya. So the user can always have a single entry point!!! This is very clever!